src/main/java/zpi/controller/UserController.java

```diff
@@ -0,0 +1,61 @@
+package zpi.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import zpi.dto.NewUserDto;
+import zpi.entity.User;
+import zpi.service.UserService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<User>> getUsers() {
+        return ResponseEntity.ok(userService.getUsers());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<User> getUser(@PathVariable final int id) {
+        return ResponseEntity.ok(userService.getUser(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<User> postUser(@RequestBody final NewUserDto user) {
+        userCheck(user);
+        return ResponseEntity.ok(userService.addUser(user));
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("/{id}")
+    public ResponseEntity<User> deleteUser(@PathVariable int id) {
+        return ResponseEntity.status(201).body(userService.deleteUser(id));
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/{id}")
+    public ResponseEntity<Void> updateUser(@PathVariable int id, @RequestBody User updatedUser) {
+        userService.updateUser(id, updatedUser);
+        return ResponseEntity.status(204).build();
+    }
+
+    private void userCheck(NewUserDto user){
```

Comment:
Kontroler to nie jest odpowiednia warstwa na takie rzeczy. Sprawdzajmy takie coś w serwisie albo dodajmy walidacje encji

Reply:
Przenioslem serwisu i troche zreformatowalem bo bylo to milion funkcjonalnosci w 1 metodzie